Client/core/CMessageLoopHook.cpp

```diff
@@ -124,15 +124,34 @@ LRESULT CALLBACK CMessageLoopHook::ProcessMessage(HWND hwnd, UINT uMsg, WPARAM w
     // Alternate alt-tab system
     if (pThis && hwnd == pThis->GetHookedWindowHandle())
     {
-        if (uMsg == WM_ACTIVATE && LOWORD(wParam) == WA_ACTIVE)
+        if (uMsg == WM_ACTIVATE)
         {
-            GetVideoModeManager()->OnGainFocus();
-        }
-        if (uMsg == WM_ACTIVATE && LOWORD(wParam) == WA_INACTIVE)
-        {
-            GetVideoModeManager()->OnLoseFocus();
-            g_pCore->GetKeyBinds()->OnLoseFocus();
+            CModManager* pModManager = CModManager::GetSingletonPtr();
+
+            if (pModManager && pModManager->IsLoaded())
+            {
+                CClientBase* pBase = pModManager->GetCurrentMod();
+
+                if (pBase)
+                {
+                    WORD state = LOWORD(wParam);
+                    bool focus = (state == WA_CLICKACTIVE) || (state == WA_ACTIVE);
+
+                    pBase->OnWindowFocusChange(focus);
```

Comment:
I believe variable naming should be consistent, so let's use Hungarian notation here as well.

Additionally, I believe the `state` variable could be moved up so that you can use it on e.g. line 144 and 148 as well, thus reducing the amount of repeated code for the same thing especially since it doesn't return here.

```suggestion
                    WORD wState = LOWORD(wParam);
                    bool bFocus = (wState == WA_CLICKACTIVE) || (wState == WA_ACTIVE);

                    pBase->OnWindowFocusChange(bFocus);
```